x-pack/plugin/sql/src/main/java/org/elasticsearch/xpack/sql/optimizer/Optimizer.java

```diff
@@ -842,17 +846,43 @@ protected LogicalPlan rule(LogicalPlan p) {
         }
     }
 
-    /**
-     * A COUNT in a local relation will always be 1.
-     */
-    private static class ReplaceCountInLocalRelation extends OptimizerRule<Aggregate> {
+    static class ReplaceAggregationsInLocalRelations extends OptimizerRule<UnaryPlan> {
```

Comment:
This is just a rename from `ReplaceCountInLocalRelation` to `ReplaceAggregationsInLocalRelations` to cover some more generic replacements.